api/openapi_server/controllers/auth_controller.py

```diff
@@ -111,6 +111,9 @@ def signUpHost():  # noqa: E501
           SecretHash=secret_hash,
           Username=body['email'],
           Password=body['password'],
+          ClientMetadata={
+              'url': ROOT_URL
```

Comment:
Since the `ROOT_URL` is a requirement for properly running `openapi_server`, we should probably put assertions up above where we read it `ROOT_URL=env.get('ROOT_URL')` - to ensure that the env var isn't empty. 

This will help devs troubleshoot errors early, and will help provide some sort of documentation around the required env vars.

Reply:
@Joshua-Douglas Thanks for the suggested changes! I added a check for the ROOT_URL variable at the top of the file. I think I'll leave the session rollback for another smaller PR. 

Regarding the sign up routes, we had a conversation a while ago and felt it was a good idea to split them up since the coordinator and host sign up processes could look quite different once we flesh out the app some more. For example, having a starting application status, which role they have, etc. and wanted to avoid a bunch of if statements based on the type of user that is singing up. Do you think we were getting ahead of ourselves here?

Reply:
Hey @erikguntner. The improved validation looks great! Thanks for doing that.

Ah, yeah I see your point about wanting to split up the sign up routes. I agree that splitting up the endpoints is a good idea, since this impacts the logic flow of the frontend app.

I think we can have the best of both words if we keep both endpoints, but update each endpoint to just call a generic `signUp` method. This way we aren't duplicating any code, and more custom implementations can be easily added in the future. 

That said, this is probably out of scope for this PR and just a minor comment